codegenerator/cli/src/config_parsing/system_config.rs

```diff
@@ -723,7 +723,17 @@ impl SystemConfig {
                     enable_raw_events: evm_config.raw_events.unwrap_or(false),
                     preload_handlers: evm_config.preload_handlers.unwrap_or(false),
                     lowercase_addresses: match evm_config.address_format {
-                        Some(super::human_config::evm::AddressFormat::Lowercase) => true,
+                        Some(super::human_config::evm::AddressFormat::Lowercase) => {
+                            // Validate that USE_HYPERSYNC_CLIENT_DECODER is not set to false when address_format is lowercase
+                            if let Ok(use_hypersync_decoder) =
+                                std::env::var("USE_HYPERSYNC_CLIENT_DECODER")
```

Comment:
Let's delete the environment variable and use only `event_decoder` option from the config. If users want, they'd be able to pass an env var to the `config.yaml`